### PR TITLE
[#788] Model deployment resources

### LINKED ISSUE #788: Resources of model deployments
Currently, we don't set the deployment resources for a model. Need to fix it.

### PR DESCRIPTION
**Pull request by mcdoker18**
Added resources for deployment. These values can be overridden through
environment variables, profile values or the parameters of deploy HTTP
query.
This closes #788

### CODE REVIEW

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

File: legion/legion/k8s/utils.py [hunk 1]
```diff
@@ -56,6 +56,60 @@
...
+    :param value: k8s resource
+    :type value: int
+    :return: reduced k8s resource - int
+    """
+    return value * (100 - int(os.getenv(*legion.config.REDUCE_MODEL_REQUESTS_BY))) // 100
+
+
+def reduce_mem_resource(res: str):
+    """
+    Reduce memory k8s resource by 30%
```
Comment (botsman) [inline legion/legion/k8s/utils.py]: Minor comment

Maybe it is better not to hardcode the exact value in docstring? 
We can reference a variable in docs which is responsible for the percent.
And it is actually 33%, right?

**Comment by kirillmakhonin** [comment]
Should be updated to new configuration system after merging of #595

File: legion/legion/k8s/utils.py (hunk 1 above)
Comment (mcdoker18) [reply legion/legion/k8s/utils.py]: Fixed

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot. test all

**Review by alexander-semenets — DISMISSED** [review]

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test all

File: legion/legion/config.py [hunk 2]
```diff
@@ -536,10 +536,11 @@ def __new__(cls, name, default=None, cast_func=str, description=None, configurab
...
 
-MODEL_VERSION = ConfigVariableDeclaration('MODEL_VERSION', None, str,
-                                          'ID of current model',
-                                          False)
+MODEL_VERSION = ConfigVariableDeclaration('MODEL_VERSION', None, str, 'ID of current model', False)
+
+MODEL_K8S_MEMORY = ConfigVariableDeclaration('MODEL_K8S_MEMORY', '256Mi', str, 'Default k8s memory for a model', True)
+MODEL_K8S_CPU = ConfigVariableDeclaration('MODEL_K8S_CPU', '256m', str, 'Default k8s cpu for a model', True)
+REDUCE_MODEL_REQUESTS_BY = ConfigVariableDeclaration('REDUCE_MODEL_REQUESTS_BY', 33, int,
+                                                     'Reducy k8s resource for model by specific percent', True)
```
Comment (kirillmakhonin) [inline legion/legion/config.py]: Typo in word `Reducy`

**Review by kirillmakhonin — DISMISSED** [review]

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test all

**Comment by mcdoker18** [comment]
bot, test app

**Comment by mcdoker18** [comment]
bot, test app

**Review by alexander-semenets — APPROVED** [review]

**Review by botsman — APPROVED** [review]

**Review by kirillmakhonin — APPROVED** [review]

**Pull request merged**